### PR TITLE
LANG: python 3.9 update (metagenome)

### PR DESCRIPTION
**Pull request by lizgehret**

### CODE REVIEW

File: 2024.5/metagenome/passed/seed-environment-conda.yml [hunk 1]
```diff
@@ -4,17 +4,18 @@ channels:
 - bioconda
 - defaults
 dependencies:
-- biom-format=2.1.14
+- biom-format=2.1.15
 - bowtie2=2.5.1
-- h5py=3.8.0
+- h5py=3.11.0
 - hdf5=1.14.0
```
Comment (ebolyen) [inline 2024.5/metagenome/passed/seed-environment-conda.yml]:
```suggestion
- hdf5=1.14.3
```
Looks like this was change was used in the amplicon distro and is related to the solve error

Comment (lizgehret) [reply 2024.5/metagenome/passed/seed-environment-conda.yml]: i think we've got at least a few things that need to be updated in order to get a solve. i'm going to do a local install and work through all of the external deps for metagenome-specific plugins to see where that lands us.

**Comment by ebolyen** [comment]
This seed env sans-q2 solves locally, there must be some conflict in the q2 conda recipes, but not sure where.

What I would probably do next is install the seed-environment without q2. Then using the rebuilt channel from the last run, install plugins one by one into the seed-sans-q2 env and see which one causes the solver to choke. A pain, but should be locally reproducible.

**Comment by lizgehret** [comment]
> This seed env sans-q2 solves locally, there must be some conflict in the q2 conda recipes, but not sure where.
> 
> What I would probably do next is install the seed-environment without q2. Then using the rebuilt channel from the last run, install plugins one by one into the seed-sans-q2 env and see which one causes the solver to choke. A pain, but should be locally reproducible.

yep, we're on the same page! 👍

**Comment by lizgehret** [comment]
Did a local install of an environment file containing:
- external deps (non-q2) in seed env from amplicon distro (with one change for hdf5 version)
- all versioned deps from metagenome-specific plugins (assembly, fondue, moshpit, sapienns)

Then started installing Q2 plugins one-by-one in their generational order. All plugins install successfully without version conflicts except for:
- q2-assembly
- q2-moshpit

The following dependencies are in conflict with the existing/required environment (present in both assembly and moshpit):

**downgrade**
r-base 4.3.3 ->  4.3.1

**upgrade**
eggnog-mapper 2.1.10 -> 2.1.12
entrezpy 2.1.2 -> 2.1.3
notebook 6.5.5 -> 7.0.8
r-reticulate 1.36.0 -> 1.36.1
tqdm 4.62.3 -> 4.66.2

(Note that these are all external packages with pins set from other plugins in the existing environment)

- Additionally, assembly and moshpit have different requirements for the version of busco. Moshpit pins busco at 5.5.0 and assembly requires an upgrade to 5.6.1.

- The final conflict is also within moshpit, which requires the following upgrade of sepp: 4.4.0 -> 4.5.1.

Will need to play around with moshpit and assembly locally on Monday to see what we can do to fix these conflicts, but leaving this here for posterity.

**Comment by lizgehret** [comment]
Closing in favor of [Prepare PR](https://github.com/qiime2/distributions/pull/249).

**Pull request closed without merging**